agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +169,29 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) installerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+	installScript, uninstallScript, err := r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret.Name, byoHost.Spec.InstallationSecret.Namespace)
+	if err != nil {
+		logger.Error(err, "error getting installation script")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "ReadInstallationSecretFailed", "installation secret %s not found", byoHost.Spec.InstallationSecret.Name)
+		return ctrl.Result{}, err
+	}
+	byoHost.Spec.UninstallationScript = &uninstallScript
+	installScript, err = r.parseInstallationScript(ctx, installScript)
```

Comment:
We have a `ParseTemplate` helper. You can reuse that - https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/b09a31d78eb0b2e824f3f98dad96f2ead0f638a8/agent/cloudinit/template_parser.go#L21-L36